strategies/categorized/src/main/java/org/arquillian/smart/testing/strategies/categorized/TagsParser.java

```diff
@@ -0,0 +1,107 @@
+package org.arquillian.smart.testing.strategies.categorized;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.arquillian.smart.testing.logger.Log;
+import org.arquillian.smart.testing.logger.Logger;
+
+class TagsParser extends AbstractParser {
+
+    public static final String TAG = "org.junit.jupiter.api.Tag";
+    public static final String TAGS = "org.junit.jupiter.api.Tags";
+    private final Logger logger = Log.getLogger();
+
+    TagsParser(CategorizedConfiguration strategyConfig) {
+        super(strategyConfig);
+    }
+
+    @Override
+    protected List<String> findCategories(Class<?> clazz) {
+
+        return Arrays.stream(clazz.getAnnotations())
+            .map(annotation -> this.findJUnit5TagAnnotation(annotation))
+            .filter(Objects::nonNull)
+            .flatMap(this::retrieveTagFromAnnotation)
+            .map(this::changeIfNonCaseSensitive)
+            .collect(Collectors.toList());
+    }
+
+    @Override
+    protected boolean isSpecified(String category, List<String> specifiedCategories) {
+        return specifiedCategories.contains(category);
+    }
+
+    private Stream<String> retrieveTagFromAnnotation(Annotation tagsAnnotation) {
+        final List<String> tags = new ArrayList<>();
+        if (TAG.equals(tagsAnnotation.annotationType().getName())) {
+            tags.add(getTagName(tagsAnnotation));
+        } else {
+            tags.addAll(getMultipleTagNames(tagsAnnotation));
+        }
```

Comment:
I can see `getTagName` and `getMultipleTagNames` returning null. I think we should check null value before adding it to list.

Reply:
Only `getTagName` can return null although is unlikely that happens because of the `if` protection, it is better to protect for just in case.